tests/tests_pytorch/trainer/logging_/test_logger_connector.py

```diff
@@ -506,6 +510,74 @@ def _assert_called(model, fn, stage):
     _assert_called(model, "test", "test")
 
 
+@pytest.mark.parametrize("compute_groups", [True, False])
+def test_metriccollection_compute_groups(tmpdir, compute_groups):
+    def assertion_calls(keep_base: bool, copy_state: bool):
+        if _TORCHMETRICS_GREATER_EQUAL_0_9_1:
+            assert copy_state != compute_groups
+
+        assert not keep_base
+
+    class CustomMetricsCollection(MetricCollection):
+        wrapped_assertion_calls = Mock(wraps=assertion_calls)
+
+        def items(self, keep_base: bool = False, copy_state: bool = True):
+            if getattr(self, "_is_currently_logging", False):
+                self.wrapped_assertion_calls(keep_base, copy_state)
+
+            return super().items(keep_base=keep_base, copy_state=copy_state)
+
+    class DummyModule(LightningModule):
+        def __init__(self):
+            super().__init__()
+            if compare_version("torchmetrics", operator.ge, "0.10.0"):
```

Comment:
What's the beenfit of having this? Why not just import a metric that is present on all versions?

Reply:
We need metrics that actually can share a compute group (which are mainly the classification ones and they have all been revamped for 0.10.0).